app/src/main/java/com/github/studydistractor/sdp/ui/CreateUserScreen.kt

```diff
@@ -159,27 +163,28 @@ fun CreateAccountScreen(
 
         // Validate
         Button( onClick = {
-            var birthday = HashMap<String,Int>()
-
-            if (selectedDateText.isNotEmpty()){
-                birthday = hashMapOf(
-                    "year" to year,
-                    "month" to month,
-                    "day" to dayOfMonth
-                )
-            }
-
             var userIsCreated = true
-            try {
-                val usr = createAccount.createUser(birthday, firstname, lastname, phone)
-            } catch (e: java.lang.IllegalArgumentException) {
+            val birthday = hashMapOf(
+                "year" to year, "month" to month, "day" to dayOfMonth)
+            val user = UserData(userId, firstname, lastname, phone, birthday, hashMapOf(), 0)
+            try{
```

Comment:
Maybe we should avoid using `try` here. Instead use callbacks : `onSuccess` and `onFailure`. The app should never crash and inputs errors should just reset the fields and give feedback to the user.

Reply:
[See](https://github.com/StudyDistractor/sdp/pull/101), I believe Athena did a good job there for handling errors

Reply:
I partly agree. If you are correctly coding `try` block, it does not crash. In this case, the function `processUser` should raise an exception if the data is wrongly initialized